cl/assets/react/DisclosureViewer.tsx

```diff
@@ -0,0 +1,463 @@
+import React from 'react';
+import { appFetch } from './_fetch';
+import { Table } from 'react-bootstrap';
+import './disclosure-page.css';
+import { useParams } from 'react-router-dom';
+import { disclosureModel } from './_disclosure_models';
+import { convertTD, fetch_year_index, getIndex } from './_disclosure_helpers';
+import debounce from 'lodash.debounce';
+import {on} from "cluster";
+
+interface TableNavigationInnerProps {
+  disclosures: string;
+}
+
+interface Person {
+  id: number;
+  name_first: string;
+  name_last: string;
+  slug: string;
+}
+
+interface Row {
+  id: number;
+  filepath: string;
+  person: Person;
+  year: string;
+  thumbnail: string;
+  page_number: number;
+  redacted: boolean;
+}
+
+interface Data {
+  has_been_extracted: boolean;
+  addendum_content_raw: string;
+  filepath: string;
+  thumbnail: string;
+  id: number;
+}
+
+const TableNavigation: React.FC<TableNavigationInnerProps> = (disclosures) => {
+  return <React.Fragment>{MainSection(disclosures)}</React.Fragment>;
+};
+
+
+
+const MainSection = (disclosures) => {
+  const years = disclosures['years'].split(',');
+  const doc_ids = disclosures['ids'].split(',');
+  const { disclosure_id } = useParams();
+  const { judge_id } = useParams();
+  const [data, setData] = React.useState('');
+  const [judge, setJudge] = React.useState([]);
+  const judge_name = disclosures['judge'];
+  const is_admin = disclosures['admin'] == 'True';
+  const indx = getIndex(disclosure_id, doc_ids);
+  const [visible, setVisible] = React.useState(true);
+
+  const fetchDisclosure = async (doc_id: number) => {
+    try {
+      const response = await appFetch(`/api/rest/v3/financial-disclosures/?person=${judge_id}&id=${doc_id}`);
+      console.log(response);
+      setData(response['results'][0]);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  if (data == '') {
+    fetchDisclosure(disclosure_id);
+  }
+
+  const fetchJudge = async (query: string) => {
+    try {
+      const response: any = await appFetch(`/api/rest/v3/disclosure-typeahead/?fullname=${query}&order_by=name_last`);
+      setJudge(response['results']);
+    } catch (error) {
+      setJudge([]);
+    }
+  };
+
+  const changeHandler = (event: string) => {
+    fetchJudge(event);
+  };
+
+  const debounceFetchJudge = React.useMemo(() => debounce(changeHandler, 300), []);
+  const urlList = window.location.pathname.split('/');
+  const judgeUrl = [urlList[0], urlList[1], urlList[2], urlList[5], urlList[6]].join('/');
```

Comment:
This hurts. Please make the API you wish to see in the world. In other words, add an attribute for the `person_absolute_url` to the FDs API so this hackery isn't needed. Ouch.